src/Libraries/RevitNodes/Elements/Element.cs

```diff
@@ -462,17 +462,19 @@ public object[] Geometry()
 
             foreach (var geometryObject in InternalGeometry())
             {
-                try
+                var geoObj = geometryObject.Convert();
+                if (geoObj != null)
                 {
-                    var convert = geometryObject.Convert();
-                    if (convert != null)
-                    {
-                        converted.Add(convert);
-                    }
+                    converted.Add(geoObj);
                 }
-                catch (Exception)
+                else
                 {
-                    // we catch all geometry conversion exceptions
+                    var solid = geometryObject as Autodesk.Revit.DB.Solid;
+                    if (solid != null)
+                    {
+                        var geomObjs = solid.ConvertToMany();
+                        converted.AddRange(geomObjs.Where(x => { return x != null; }));
+                    }
```

Comment:
I put the try catch inside of the foreach so that you'd get as much geometry out as is possible. This isn't ideal obviously, but it gives users a "best effort" solution. Do you disagree with that?